### PR TITLE
Enables pagination for Saves & Follows views

### PR DESCRIPTION
**Pull request by l2succes**
Fixes: https://github.com/artsy/collector-experience/issues/729

Skip New Tests

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Missing Test Files:

- `src/lib/Scenes/Favorites/Components/Artists/__tests__/index-tests.tsx`
- `src/lib/Scenes/Favorites/Components/Artworks/__tests__/index-tests.tsx`
- `src/lib/Scenes/Favorites/Components/Categories/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>




<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Scenes/Favorites/Components/Artists/__tests__/index-tests.tsx`
- `src/lib/Scenes/Favorites/Components/Artworks/__tests__/index-tests.tsx`
- `src/lib/Scenes/Favorites/Components/Categories/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Scenes/Favorites/Components/Artists/__tests__/index-tests.tsx`
- `src/lib/Scenes/Favorites/Components/Artworks/__tests__/index-tests.tsx`
- `src/lib/Scenes/Favorites/Components/Categories/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

File: src/lib/Scenes/Favorites/Components/Artworks/index.tsx [hunk 1]
```diff
@@ -23,7 +51,14 @@ export class SavedWorks extends React.Component<any, any> {
     )
 
     const Content = (
-      <Container>
+      <Container
+        onScroll={({ nativeEvent }) => {
+          if (this.isCloseToBottom(nativeEvent)) {
```
Comment (damassi) [inline src/lib/Scenes/Favorites/Components/Artworks/index.tsx]: :)

File: src/lib/Scenes/Favorites/Components/Artworks/index.tsx [hunk 2]
```diff
@@ -23,7 +51,14 @@ export class SavedWorks extends React.Component<any, any> {
...
+            this.loadMore()
```
Comment (damassi) [inline src/lib/Scenes/Favorites/Components/Artworks/index.tsx]: :) One thing i noticed when i was building this out in my other PR is that when it gets close its really sensitive and fires a ton of times; to fix it I wrapped my call in `_.once` which basically only fires 1 time per render cycle. Not sure how much this applies in react-native context tho

Comment (l2succes) [reply src/lib/Scenes/Favorites/Components/Artworks/index.tsx]: Oh that's possible, I have a check in `loadMore` so it only sends one request for data at a time

Comment (l2succes) [reply src/lib/Scenes/Favorites/Components/Artworks/index.tsx]: Since we have a util for it now, I'll just go ahead and use it here as well

**Comment by ArtsyOpenSource** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  Missing Test Files:

- `src/lib/Scenes/Favorites/Components/Artists/__tests__/index-tests.tsx`
- `src/lib/Scenes/Favorites/Components/Artworks/__tests__/index-tests.tsx`
- `src/lib/Scenes/Favorites/Components/Categories/__tests__/index-tests.tsx`
- `src/lib/Scenes/Favorites/__tests__/index-tests.tsx`

If these files are supposed to not exist, please update your PR body to include "Skip New Tests".
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Pull request merged**